From ffc53b5e7dff85cdd5cec48a77fac49a87b44c52 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 7 Jun 2016 11:09:45 +0300
Subject: [PATCH 0490/2241] fix: i2c: update buffer array type to be u8,
 instead of u32

Fix the buffer array type of messages to be u8 instead of u32.
Issue found when enable I2C bridge driver in Big endian Kernel
This fix is relevant for I2C-bridge driver only

Change-Id: Idafdc787e004a4348dcd77438ea06a1c04f1ad3e
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30350
---
 drivers/i2c/busses/i2c-mv64xxx.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/i2c/busses/i2c-mv64xxx.c b/drivers/i2c/busses/i2c-mv64xxx.c
index ef7b35d..7a8e818 100644
--- a/drivers/i2c/busses/i2c-mv64xxx.c
+++ b/drivers/i2c/busses/i2c-mv64xxx.c
@@ -434,10 +434,10 @@ static void
 mv64xxx_i2c_read_offload_rx_data(struct mv64xxx_i2c_data *drv_data,
 				 struct i2c_msg *msg)
 {
-	u32 buf[2];
+	u8 buf[2];
 
-	buf[0] = readl(drv_data->reg_base + MV64XXX_I2C_REG_RX_DATA_LO);
-	buf[1] = readl(drv_data->reg_base + MV64XXX_I2C_REG_RX_DATA_HI);
+	buf[0] = readb(drv_data->reg_base + MV64XXX_I2C_REG_RX_DATA_LO);
+	buf[1] = readb(drv_data->reg_base + MV64XXX_I2C_REG_RX_DATA_HI);
 
 	memcpy(msg->buf, buf, msg->len);
 }
@@ -592,12 +592,12 @@ static void
 mv64xxx_i2c_prepare_tx(struct mv64xxx_i2c_data *drv_data)
 {
 	struct i2c_msg *msg = drv_data->msgs;
-	u32 buf[2];
+	u8 buf[2];
 
 	memcpy(buf, msg->buf, msg->len);
 
-	writel(buf[0], drv_data->reg_base + MV64XXX_I2C_REG_TX_DATA_LO);
-	writel(buf[1], drv_data->reg_base + MV64XXX_I2C_REG_TX_DATA_HI);
+	writeb(buf[0], drv_data->reg_base + MV64XXX_I2C_REG_TX_DATA_LO);
+	writeb(buf[1], drv_data->reg_base + MV64XXX_I2C_REG_TX_DATA_HI);
 }
 
 static int
-- 
2.7.4

